File: examples/example-64/ruby/features/step_definitions/screenplay.rb
Given('the homepage is opened with screenplay') do
    @actor.performs(Tasks::OpenBrowser.new)
end

When('{string} is typed into the search box with screenplay') do |value|
    @actor.performs(Tasks::TypeSearch.new(value))
end

When('the {string} button is pressed with screenplay') do |btn|
    @actor.performs(Tasks::SubmitSearch.new)
end

Then('the results should display {string} with screenplay') do |results|
    @actor.asks(Questions::ResultsText.new(results))
end

File: examples/example-64/js/features/step_definitions/screenplay.js
const { Given, When, Then } = require('@cucumber/cucumber');
const { Navigate, Enter, Click, Text, PageElement, By } = require('@serenity-js/web');
const { Ensure, includes } = require('@serenity-js/assertions');

Given('the homepage is opened with screenplay', function () {
    return this.actor.attemptsTo(
        Navigate.to('http://localhost:9999')
    );
});

When('{string} is typed into the search box with screenplay', function (text) {
    return this.actor.attemptsTo(
        Enter.theValue(text).into(PageElement.located(By.css('input[type="text"]')))
    );
});

When('the {string} button is pressed with screenplay', function (btn) {
    return this.actor.attemptsTo(
        Click.on(PageElement.located(By.css(`input[type="submit"][value="${btn}"]`)))
    );
});

Then('the results should display {string} with screenplay', function (expectedText) {
    return this.actor.attemptsTo(
        Ensure.that(
            Text.of(PageElement.located(By.css('div#results'))),
            includes(expectedText))
    );
});